language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/ExMemory.hs

```diff
@@ -90,20 +92,28 @@ instance (Generic a, GExMemoryUsage (Rep a)) => ExMemoryUsage (GenericExMemoryUs
 class ExMemoryUsage a where
     memoryUsage :: a -> ExMemory -- ^ How much memory does 'a' use?
 
-deriving via (GenericExMemoryUsage (Constant ann)) instance ExMemoryUsage ann => ExMemoryUsage (Constant ann)
 deriving via (GenericExMemoryUsage (Name ann)) instance ExMemoryUsage ann => ExMemoryUsage (Name ann)
-deriving via (GenericExMemoryUsage TypeBuiltin) instance ExMemoryUsage TypeBuiltin
-deriving via (GenericExMemoryUsage (Type TyName ann)) instance ExMemoryUsage ann => ExMemoryUsage (Type TyName ann)
+deriving via (GenericExMemoryUsage (Type TyName uni ann)) instance ExMemoryUsage ann => ExMemoryUsage (Type TyName uni ann)
 deriving via (GenericExMemoryUsage (Builtin ann)) instance ExMemoryUsage ann => ExMemoryUsage (Builtin ann)
 deriving via (GenericExMemoryUsage (Kind ann)) instance ExMemoryUsage ann => ExMemoryUsage (Kind ann)
 deriving via (GenericExMemoryUsage BuiltinName) instance ExMemoryUsage BuiltinName
 deriving via (GenericExMemoryUsage DynamicBuiltinName) instance ExMemoryUsage DynamicBuiltinName
-deriving via (GenericExMemoryUsage (Plain Term)) instance ExMemoryUsage (Plain Term)
-deriving via (GenericExMemoryUsage (WithMemory Term)) instance ExMemoryUsage (WithMemory Term)
+-- TODO: Can we merge these two instances together?
```

Comment:
Never actually tried, may be possible with expanding the synonyms.

Reply:
This works:

```haskell
deriving via (GenericExMemoryUsage (Term TyName Name uni ann))
  instance (ExMemoryUsage ann, Closed uni, uni `Everywhere` ExMemoryUsage) =>
    ExMemoryUsage (Term TyName Name uni ann)
```

Looks fine?

Reply:
Looks reasonable.